### PR TITLE
Adds SOL Protocol System

### PR DESCRIPTION
**Pull request by ShibberinoShipperino**
### **Adds new Protocol System for Security Levels:**

_Code Green - Daybreak Protocol
Code Blue - Sunset Protocol
Code Red - Nightfall Protocol
Code Delta - Supernova Protocol (Uh Oh)_

This also changes various computers to accommodate the new levels.

Config Files have been changed to reflect the new announcement system.

:cl: Laniakeaa
 * rscadd: Adds a new Protocol System for Crew. Read up!
 * rscadd: Adds new buttons/ui to reflect Protocol Systems on Computers/Keycard Authenticator.

### CODE REVIEW

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -362,7 +362,7 @@ var/global/ports_open = TRUE
 						ports_open = TRUE
 						return
 				else
-					to_chat(usr, "<span class='warning'>This action requires either a red alert or head of security authorization.</span>")
+					to_chat(usr, "<span class='warning'>This action requires either a Nightfall Protocol active or head of security authorization.</span>")
```
Comment (msgerbs) [inline code/game/machinery/computer/communications.dm]: Grammar here is slightly clunky, suggest revising.

File: code/game/machinery/computer/communications.dm [hunk 2]
```diff
@@ -411,17 +411,17 @@ var/global/ports_open = TRUE
 		),
 		"alerts"=list(
 			list("alert"="default",   "label"="Nanotrasen",  "desc"="Oh god."),
-			list("alert"="redalert",  "label"="Red Alert",   "desc"="Nothing to do with communists."),
+			list("alert"="nightfall",  "label"="Nightfall protocol",   "desc"="Nothing to do with communists."),
```
Comment (msgerbs) [inline code/game/machinery/computer/communications.dm]: This joke doesn't make much sense anymore, let's change this.

File: code/modules/security levels/security levels.dm [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
 
 	//Will not be announced if you try to set to the same level as it already is
-	if(level >= SEC_LEVEL_GREEN && level <= SEC_LEVEL_DELTA && level != security_level)
+	if(level >= SEC_LEVEL_DAYBREAK && level <= SEC_LEVEL_SUPERNOVA && level != security_level)
 		switch(level)
-			if(SEC_LEVEL_GREEN)
+			if(SEC_LEVEL_DAYBREAK)
 				world << sound('sound/misc/notice2.ogg')
-				to_chat(world, "<font size=4 color='red'>Attention! Security level lowered to green</font>")
+				to_chat(world, "<font size=4 color='red'>Attention! Sunset Protocol has been lowered to Daybreak.</font>")
```
Comment (msgerbs) [inline code/modules/security levels/security levels.dm]: This also feels a bit clunky. What's the direct replacement for the general concept of "security level" in this system? Alternatively, I suggest rephrasing the whole message.

File: config-example/config.txt [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
-# SERVERNAME tgstation
+# SERVERNAME Derelict 12
 
 ## Alert levels
-ALERT_GREEN All threats to the station have passed. Security may not have weapons visible, privacy laws are once again fully enforced.
-ALERT_BLUE_UPTO The station has received reliable information about possible hostile activity on the station. Security staff may have weapons visible, random searches are permitted.
-ALERT_BLUE_DOWNTO The immediate threat has passed. Security may no longer have weapons drawn at all times, but may continue to have them visible. Random searches are still allowed.
-ALERT_RED_UPTO There is an immediate serious threat to the station. Security may have weapons unholstered at all times. Random searches are allowed and advised.
-ALERT_RED_DOWNTO The self-destruct mechanism has been deactivated, there is still however an immediate serious threat to the station. Security may have weapons unholstered at all times, random searches are allowed and advised.
+ALERT_GREEN All Daybreak protocol reassigned. No immediate threats detected, crew may continue to carry out regular work. Sunset Division are to return to standard posts and await further instruction.
```
Comment (msgerbs) [inline config-example/config.txt]: Did you forget the delete the "all" at the beginning?

File: config-example/config.txt [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
+ALERT_BLUE_DOWNTO Sunset has returned. Nightfall protocols are rescinded. Please continue your shift with caution, and comply with Asset Protection's requests.
+ALERT_RED_UPTO Nightfall protocol initiated, immediate threat confirmed. Crew rights have been limited, all personnel are to be secluded to their respective departments. Failure to comply will result in jail sentence and immediate relief of duty! This is not a drill.
+ALERT_RED_DOWNTO Supernova procedure has been averted; standard law protocol has been reinstated. All crew are to return to their departments or proceed with evacuation in an orderly manner..
 ALERT_DELTA The station's self-destruct mechanism has been engaged. All crew are instructed to obey all instructions given by heads of staff. Any violations of these orders can be punished by death. This is not a drill.
 
 ## Add a # infront of this if you want to use the SQL based admin system, the legacy system uses admins.txt. You need to set up your database to use the SQL based system.
-ADMIN_LEGACY_SYSTEM
+ADMIN_LEGACY_SYSTEMS
```
Comment (msgerbs) [inline config-example/config.txt]: What??

File: config-example/config.txt (hunk 4 above)
Comment (msgerbs) [inline config-example/config.txt]: These config values should all probably be renamed to reflect the new system.

File: config-example/config.txt (hunk 5 above)
Comment (ShibberinoShipperino) [reply config-example/config.txt]: even im confused how i did that

File: config-example/config.txt [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
-ALERT_RED_UPTO There is an immediate serious threat to the station. Security may have weapons unholstered at all times. Random searches are allowed and advised.
-ALERT_RED_DOWNTO The self-destruct mechanism has been deactivated, there is still however an immediate serious threat to the station. Security may have weapons unholstered at all times, random searches are allowed and advised.
-ALERT_DELTA The station's self-destruct mechanism has been engaged. All crew are instructed to obey all instructions given by heads of staff. Any violations of these orders can be punished by death. This is not a drill.
+ALERT_DAYBREAK Daybreak protocol reassigned. No immediate threats detected, crew may continue to carry out regular work. Sunset Division are to return to standard posts and await further instruction.
+ALERT_SUNSET_UPTO Sunset protocol initiated. Possible or minimal threat detected, Any and all suspicious activity should be reported to security. Crew may carry out regular work with maximum sensors activated, Extra vehicular activity is prohibited.
```
Comment (MrFroztee) [inline config-example/config.txt]: "Sunset protocol initiated. Possible or minimal threat detected, any and all suspicious activity should be reported to security. Crew may carry out regular work, extra vehicular activity is prohibited."

File: config-example/config.txt [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
+ALERT_SUNSET_DOWNTO Sunset has returned. Nightfall protocols are rescinded. Please continue your shift with caution, and comply with Asset Protection's requests.
```
Comment (MrFroztee) [inline config-example/config.txt]: "Sunset has returned. As such, Nightfall protocols are rescinded. Please continue the shift with caution, and comply with Asset Protection where necessary."

File: config-example/config.txt [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
+ALERT_NIGHTFALL_UPTO Nightfall protocol initiated, immediate threat confirmed. Crew rights have been limited, all personnel are to be secluded to their respective departments. Failure to comply will result in jail sentence and immediate relief of duty! This is not a drill.
+ALERT_NIGHTFALL_DOWNTO Supernova procedure has been averted; standard law protocol has been reinstated. All crew are to return to their departments or proceed with evacuation in an orderly manner..
```
Comment (MrFroztee) [inline config-example/config.txt]: "Supernova procedure has been averted; standard Nightfall protocol has been reinstated. All crew are to return to their departments or proceed with evacuation in an orderly manner."

File: config-example/config.txt [hunk 9]
```diff
@@ -1,13 +1,13 @@
...
-ALERT_BLUE_UPTO The station has received reliable information about possible hostile activity on the station. Security staff may have weapons visible, random searches are permitted.
-ALERT_BLUE_DOWNTO The immediate threat has passed. Security may no longer have weapons drawn at all times, but may continue to have them visible. Random searches are still allowed.
-ALERT_RED_UPTO There is an immediate serious threat to the station. Security may have weapons unholstered at all times. Random searches are allowed and advised.
-ALERT_RED_DOWNTO The self-destruct mechanism has been deactivated, there is still however an immediate serious threat to the station. Security may have weapons unholstered at all times, random searches are allowed and advised.
-ALERT_DELTA The station's self-destruct mechanism has been engaged. All crew are instructed to obey all instructions given by heads of staff. Any violations of these orders can be punished by death. This is not a drill.
+ALERT_DAYBREAK Daybreak protocol reassigned. No immediate threats detected, crew may continue to carry out regular work. Sunset Division are to return to standard posts and await further instruction.
```
Comment (Codydozer) [inline config-example/config.txt]: At the end of a complete sentence, use a semicolon or a period. "No immediate threats detected; crew may continue to carry out regular work.". It's a run-on sentence.

File: config-example/config.txt [hunk 10]
```diff
@@ -1,13 +1,13 @@
...
+ALERT_SUNSET_UPTO Sunset protocol initiated. Possible or minimal threat detected, Any and all suspicious activity should be reported to security. Crew may carry out regular work with maximum sensors activated, Extra vehicular activity is prohibited.
+ALERT_SUNSET_DOWNTO Sunset has returned. Nightfall protocols are rescinded. Please continue your shift with caution, and comply with Asset Protection's requests.
+ALERT_NIGHTFALL_UPTO Nightfall protocol initiated, immediate threat confirmed. Crew rights have been limited, all personnel are to be secluded to their respective departments. Failure to comply will result in jail sentence and immediate relief of duty! This is not a drill.
```
Comment (MrFroztee) [inline config-example/config.txt]: "Nightfall protocol initiated, immediate threat confirmed. Crew rights have been limited and all personnel are to be restricted to their respective departments. Failure to comply may result in a jail sentence or immediate relief of duty. This is not a drill!"

File: config-example/config.txt (hunk 6 above)
Comment (Codydozer) [inline config-example/config.txt]: "A possible or minimal threat has been detected. Any and all suspicious activity should be reported to the security forces. Crew may carry out regular work with maximum sensors activated; however extravehicular activity is prohibited.".

File: config-example/config.txt (hunk 7 above)
Comment (Codydozer) [inline config-example/config.txt]: > Nightfall protocols are rescinded.

Nightfall protocols rescinded.

File: config-example/config.txt (hunk 10 above)
Comment (Codydozer) [inline config-example/config.txt]: > Nightfall protocol initiated, immediate threat confirmed. Crew rights have been limited, all personnel are to be secluded to their respective departments. Failure to comply will result in jail sentence and immediate relief of duty!

Nightfall protocol initiated. Immediate threat confirmed. Crew rights have been limited; all personnel are to be secluded to their respective departments. Failure to comply will result in jail sentence and immediate relief of duty!

File: config-example/config.txt [hunk 11]
```diff
@@ -1,13 +1,13 @@
...
+ALERT_NIGHTFALL_DOWNTO Supernova procedure has been averted; standard law protocol has been reinstated. All crew are to return to their departments or proceed with evacuation in an orderly manner..
+ALERT_SUPERNOVA Supernova procedure active. Utter destruction of the station is imminent. Immediate evacuation in progress, martial law is in effect. Failure to comply to members of security will result in immediate execution. THIS IS NOT A DRILL!
```
Comment (Codydozer) [inline config-example/config.txt]: > Immediate evacuation in progress, martial law is in effect.

Immediate evacuation in progress and martial law is in effect.

**Comment by msgerbs** [comment]
@ShibberinoShipperino You need to find and replace the config vars that you changed and change them everywhere else too.

File: code/game/gamemodes/game_mode.dm [hunk 12]
```diff
@@ -241,8 +241,8 @@
 /*	for(var/mob/M in player_list)
 		if(!istype(M,/mob/new_player))
 			M << sound('sound/AI/intercept.ogg')
-	if(security_level < SEC_LEVEL_BLUE)
-		set_security_level(SEC_LEVEL_BLUE)*/
+	if(security_level < SEC_LEVEL_SUNSET)
```
Comment (Birdtalon) [inline code/game/gamemodes/game_mode.dm]: This is just commented out code. Better to just remove this - it's probably ancient.

**Comment by ShibberinoShipperino** [comment]
Just needs reviewing again, all working now.

**Pull request merged**